server.R

```diff
@@ -1135,4 +1135,63 @@ server <- function(input, output, session) {
     fig
   })
   
+  # Calculate base savings dynamically based on user inputs
+  base_savings <- reactive({
+    # Calculate total savings from manpower, pilferage, and idling
+    manpower_savings <- cost.df()$Saved[1] + cost.df()$Saved[2] + cost.df()$Saved[3] + cost.df()$Saved[4]
+    pilferage_savings <- pilferage_values()$vol_saved_yearly * 86  # Assuming ₹86 per litre
+    idling_savings <- (idle_total()$idling_all_ldp - idle_total()$idle_mod_all_consump_lpd) * 365 * 86
+    
+    total_savings <- manpower_savings + pilferage_savings + idling_savings
+    total_savings
+  })
+
+  output$total_savings_text <- renderText({
+    years <- input$years_slider
+    total_savings <- base_savings() * years  # Total savings over the selected years
+    paste("Total Projected Savings over", years, "year(s): ₹", format_indian(total_savings))
+  })
+
+  output$savings_projection_plot <- renderPlotly({
+    years <- input$years_slider
+    projected_savings <- base_savings() * (0:years)  # Cumulative savings over the years
+    
+    # Create a data frame for plotting
+    savings_data <- data.frame(
+      Year = 0:years,
+      Savings = projected_savings
+    )
+    
+    # Create the bar plot
+    gg <- ggplot(savings_data, aes(x = Year, y = Savings)) +
+      geom_bar(stat = "identity", fill = "blue") +
+      labs(title = "Projected Savings Over Time",
+           x = "Years",
+           y = "Projected Savings (₹)") +
+      theme_minimal()
+    
+    ggplotly(gg)
+  })
+
+  output$savings_comparison_plot <- renderPlotly({
+    years <- input$years_slider
+    projected_savings <- base_savings() * years  # Total projected savings for the selected years
+    
+    # Current cost (for comparison)
+    current_cost <- 10000000  # Example current cost
```

Comment:
This is still boiler plate code (likely from chatgpt).
You need to fill it up with real data as visible in the other tabs.
![Screenshot 2024-12-16 at 2 18 41 PM](https://github.com/user-attachments/assets/59dc175e-16c3-4d7b-a1e8-9c757a9927ef)

Reply:
@14deepika the above comment was for you.

Reply:
Working on it